src/lib/holocene/table/paginated-table/api-paginated.svelte

```diff
@@ -0,0 +1,229 @@
+<script lang="ts">
+  import type { HTMLAttributes } from 'svelte/elements';
+
+  import { onMount } from 'svelte';
+
+  import Alert from '$lib/holocene/alert.svelte';
+  import EmptyState from '$lib/holocene/empty-state.svelte';
+  import IconButton from '$lib/holocene/icon-button.svelte';
+  import Loading from '$lib/holocene/loading.svelte';
+  import FilterSelect from '$lib/holocene/select/filter-select.svelte';
+  import {
+    createPaginationStore,
+    type PaginationStore,
+  } from '$lib/stores/api-pagination';
+  import { options } from '$lib/stores/pagination';
+  import { isError } from '$lib/utilities/is';
+
+  import PaginatedTable from './index.svelte';
+
+  type T = $$Generic;
+  type $$Props = HTMLAttributes<HTMLDivElement> & {
+    onError?: (error: Error | unknown) => void | undefined;
+    onFetch: () => Promise<PaginatedRequest<T>>;
+    onShiftUp?: (event: KeyboardEvent) => void | undefined;
+    onShiftDown?: (event: KeyboardEvent) => void | undefined;
+    onSpace?: (event: KeyboardEvent) => void | undefined;
+    total?: string | number;
+    pageSizeSelectLabel: string;
+    emptyStateMessage: string;
+    fallbackErrorMessage: string;
+    itemsKeyname?: string;
+    previousButtonLabel: string;
+    nextButtonLabel: string;
+  };
+
+  type PaginatedRequest<T> = (
```

Comment:
Can you put this in a `<script context="module" />` and export it? It makes it handy when using paginated tables.